### PR TITLE
Omitempty on JSON decoder

### PR DESCRIPTION
**Pull request by mossid**
Closes: #261

### CODE REVIEW

**Review by alexanderbez — APPROVED** [review]

**Comment by alexanderbez** [comment]
It would be ideal to get this into the SDK before we release v0.35.0. It'll make parameter change proposals much easier to write (ie. you won't have to include the entire struct -- only the fields you want to change).

/cc @Liamsi

File: json-decode.go [hunk 1]
```diff
@@ -396,8 +396,12 @@ func (cdc *Codec) decodeReflectJSONStruct(bz []byte, info *TypeInfo, rv reflect.
 			// but perhaps we are aiming for as much compatibility here.
 			// JAE: I vote we depart from encoding/json, than carry a vuln.
 
-			// Set nil/zero on frv.
-			frv.Set(reflect.Zero(frv.Type()))
+			// Set to the zero value only if not omitemptuy
```
Comment (liamsi) [inline json-decode.go]:
```suggestion
			// Set to the zero value only if not omitempty
```

**Comment by liamsi** [comment]
ref: #261

**Review by liamsi — APPROVED** [review]
👍

**Comment by liamsi** [comment]
Changed the base branch to develop. Will cut a release containing this change later today.

**Comment by alexanderbez** [comment]
Great, thanks @Liamsi!

**Pull request merged**